lib/properties.js

```diff
@@ -13,8 +14,8 @@ class Properties extends EventEmitter {
 
     this.initialized = false;
 
-    this.layers = [];
-    this.properties = {};
+    this._layers = Immutable.List();
+    this._properties = Immutable.Map();
```

Comment:
I don't think that introducing additional complexity is the right way to solve this. Rather, we should refactor the parts that actually (unnecessarily) require usage of `reverse` in the first place, making the model simpler and more consistent with the interfaces it's supposed to present.